File: magmac/src/java/magma/api/contain/stream/EmptyHead.java
package magma.api.contain.stream;

import magma.api.option.Option;

import static magma.api.option.None.None;

public class EmptyHead<T> implements Head<T> {
    public static <T> Head<T> EmptyHead() {
        return new EmptyHead<>();
    }

    @Override
    public Option<T> head() {
        return None();
    }
}

File: src/ts/magma/api/collect/EmptyHead.ts
import { None } from "../../../magma/api/option/None";
import { Option } from "../../../magma/api/option/Option";
import { Head } from "./Head";
export class EmptyHead<T> implements Head<T> {
	next(): Option<T> {
		return new None<T>();
	}
}
